README.md
# react-magic-slider-dots

> React Magic Slider Dots Component for React Slick Carousel

##### _Inspired by Instagram_

[![NPM](https://img.shields.io/npm/v/react-magic-slider-dots.svg)](https://www.npmjs.com/package/react-magic-slider-dots) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)

![screenshot](/magic-slider-dots-screenshot.png?raw=true)

## Install

```bash
npm install --save react-magic-slider-dots
```

⚠️ Also install react-slick and slick-carousel for css and font [React Slick Documentation](https://github.com/akiran/react-slick)

```bash
npm install slick-carousel
npm install react-slick --save
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
```

## [DEMO](https://patricktran.github.io/react-magic-slider-dots/)

## Usage

```jsx
import React, { Component } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import MagicSliderDots from 'react-magic-slider-dots';
import 'react-magic-slider-dots/dist/magic-dots.css';

class App extends Component {
  render() {
    const settings = {
      dots: true,
      arrows: true,
      infinite: false,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      appendDots: dots => {
        return <MagicSliderDots dots={dots} numDotsToShow={4} dotWidth={30} />;
      }
    };

    return (
      <Slider {...settings}>
        <div>
          <h3>1</h3>
        </div>
        <div>
          <h3>2</h3>
        </div>
        <div>
          <h3>3</h3>
        </div>
        <div>
          <h3>4</h3>
        </div>
        <div>
          <h3>5</h3>
        </div>
        <div>
          <h3>6</h3>
        </div>
        <div>
          <h3>7</h3>
        </div>
        <div>
          <h3>8</h3>
        </div>
      </Slider>
    );
  }
}
```

### API

This a list of props that you can pass down to the component:

| Property                | Description                                                      | Default value           | Type   | Required |
| ----------------------- | ---------------------------------------------------------------- | ----------------------- | ------ | -------- |
| `dots`                  | array of HTML li elements representing the slider dot            |                         | array  | yes      |
| `numDotsToShow`         | number of slider dots to show                                    |                         | number | yes      |
| `dotWidth`              | width, in pixels, of a slider dot including any margins/padding  |                         | number | yes      |
| `dotContainerClassName` | class name of parent div                                         | `magic-dots slick-dots` | string |
| `activeDotClassName`    | class name of active slider dot                                  | `slick-active`          | string |
| `prevNextDotClassName`  | class name of left-most (prev) and right-most (next) slider dot. | `small`                 | string |

## License

MIT © [patricktran](https://github.com/patricktran)


CHANGELOG.md
<a name="1.2.2"></a>
# [1.2.2](https://github.com/patricktran/react-magic-slider-dots/compare/1.2.1...1.2.2) (2020-06-30)

Remove scarf

<a name="1.2.1"></a>
# [1.2.1](https://github.com/patricktran/react-magic-slider-dots/compare/1.2.0...1.2.1) (2020-05-14)

Add scarf installation anayltics

<a name="1.2.0"></a>

# [1.2.0](https://github.com/patricktran/react-magic-slider-dots/compare/v1.1.3...1.2.0) (2019-09-10)

Limited support for responsive breakpoints from react-slick  
Reported react-slick breakpoint issue: https://github.com/akiran/react-slick/issues/1642

<a name="1.1.3"></a>

# [1.1.3](https://github.com/patricktran/react-magic-slider-dots/compare/v1.1.1...v.1.1.3) (2019-07-03)

Support for infinite={true} from react-slick

<a name="1.1.2"></a>

# [1.1.1/1.1.2](2018-08-10)

Support for initialSlide value from react-slick being greater than 0
Use translateX hardware accelerated CSS
Update package.json keywords


example/src/App.js
import React, { Component } from 'react';
import Slider from 'react-slick';
import ToggleButton from 'react-toggle-button';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { tomorrowNightEighties } from 'react-syntax-highlighter/styles/hljs';
import MagicSliderDots from 'react-magic-slider-dots';
import 'react-magic-slider-dots/dist/magic-dots.css';

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      enableMagicSliderDots: true
    };
  }

  handleDotToggle = () => {
    this.setState({
      enableMagicSliderDots: !this.state.enableMagicSliderDots
    });
  };

  contentIndexes = [...Array.from({ length: 10 }, (v, k) => k + 1).map(item => item)];

  render() {
    const { enableMagicSliderDots } = this.state;

    var settings = {
      dots: true,
      arrows: true,
      infinite: false,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      responsive: [
        {
          breakpoint: 800,
          settings: {
            slidesToShow: 4
          }
        },
        {
          breakpoint: 480,
          settings: {
            slidesToShow: 2
          }
        }
      ]
    };

    if (enableMagicSliderDots)
      settings['appendDots'] = dots => {
        return <MagicSliderDots dots={dots} numDotsToShow={5} dotWidth={30} />;
      };

    return (
      <div>
        <label>
          <ToggleButton value={enableMagicSliderDots} onToggle={this.handleDotToggle} />
          &nbsp;enable magic-slider-dots
        </label>
        <Slider {...settings} key={`slider_${enableMagicSliderDots}`}>
          {this.contentIndexes.map(index => (
            <div key={index}>
              <h3>{index}</h3>
            </div>
          ))}
        </Slider>

        <section className="code-example">
          <h4>
            API -{' '}
            <a href="https://patricktran.github.io/react-magic-slider-dots/" target="_blank">
              click here
            </a>
          </h4>
          <h4>Installation</h4>
          <SyntaxHighlighter language="javascript" style={tomorrowNightEighties}>
            {`
            //install react slick and css
            npm install react-slick --save
            npm install slick-carousel --save

            //install react-magic-slider-dots
            npm install react-magic-slider-dots --save
        `}
          </SyntaxHighlighter>
          <h4></h4>
          <SyntaxHighlighter language="javascript" style={tomorrowNightEighties}>{`
        import React, { Component } from 'react'
        import Slider from 'react-slick';
        import 'slick-carousel/slick/slick.css';
        import 'slick-carousel/slick/slick-theme.css';
        import MagicSliderDots from 'react-magic-slider-dots';
        import 'react-magic-slider-dots/dist/magic-dots.css';

        class App extends Component {

        render() {

          const settings = {
              dots: true,
              arrows: true,
              infinite: false,
              speed: 500,
              slidesToShow: 1,
              slidesToScroll: 1,
              appendDots: (dots) => {
                return <MagicSliderDots dots={dots} numDotsToShow={5} dotWidth={30} />
              }
            };

            return (<Slider {...settings}>
              <div>
                <h3>1</h3>
              </div>
              <div>
                <h3>2</h3>
              </div>
              <div>
                <h3>3</h3>
              </div>
              <div>
                <h3>4</h3>
              </div>
              <div>
                <h3>5</h3>
              </div>
              <div>
                <h3>6</h3>
              </div>
            </Slider>)
          }
        }`}</SyntaxHighlighter>
        </section>
      </div>
    );
  }
}
